onnxruntime/core/session/model_compilation_options.cc

```diff
@@ -1,26 +1,136 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+
+#if !defined(ORT_MINIMAL_BUILD)
 #include "core/session/model_compilation_options.h"
 
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "core/session/allocator_adapters.h"
+#include "core/session/onnxruntime_session_options_config_keys.h"
+#include "core/session/ort_env.h"
+
 namespace onnxruntime {
+ModelCompilationOptions::ModelCompilationOptions(const OrtEnv& env, const OrtSessionOptions& session_options)
+    : env_(env), session_options_(std::make_unique<OrtSessionOptions>(session_options)) {
```

Comment:
>std::make_unique<OrtSessionOptions>(session_options) [](http://example.com/codeflow?start=34&length=52)

Since the session options do not appear to be optional, what is the purpose of allocating it on a heap?

Reply:
Ah yes. It used to be optional in a previous iteration. Now it is not, so I updated this to store the session options member by value.